Value of type variable "Self" of "__new__" of "tuple" cannot be <some subclass of typing.Tuple>

<!--
If you're not sure whether what you're experiencing is a mypy bug, please see the "Question and Help" form instead.
Please consider:

- checking our common issues page: https://mypy.readthedocs.io/en/stable/common_issues.html
- searching our issue tracker: https://github.com/python/mypy/issues to see if it's already been reported
- asking on gitter chat: https://gitter.im/python/typing
-->

**Bug Report**

<!--
If you're reporting a problem with a specific library function, the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues

If the project you encountered the issue in is open source, please provide a link to the project.
-->

Starting with version 1.1, I get an error when calling `Tuple[float, ...]`'s `__new__`.

**To Reproduce**

```python
from typing import Iterable, Tuple


class Foo(Tuple[float, ...]):
    def __new__(cls, values: Iterable[float]) -> Foo:
        return tuple.__new__(cls, map(float, values))
```

**Expected Behavior**

This should not be rejected by mypy (and it wasn't until version 1.1.1)

**Actual Behavior**

Starting with mypy 1.1.1, mypy complains with the following error
```
test.py:8: error: Value of type variable "Self" of "__new__" of "tuple" cannot be "Foo"  [type-var]
```

**Remark**
Changing `tuple.__new__` to `super().__new__` makes mypy complain (both mypy version 1.1.1 and 1.0.1) with 
```
Argument 1 to "map" has incompatible type "Type[float]"; expected "Callable[[float], _T_co]"
```

**Your Environment**

<!-- Include as many relevant details about the environment you experienced the bug in -->

- Mypy version used: 1.1.1 and 1.0.1
- Mypy command-line flags: none
- Mypy configuration options from ` pyproject.toml`:
  ```toml
  [tool.mypy]
  
  [[tool.mypy.overrides]]
  module = "pandas.*"
  ignore_missing_imports = true
  ```
- Python version used: tested on 3.8 and 3.11

<!-- You can freely edit this text, please remove all the lines you believe are unnecessary. -->

--- a/src/neo4j/_spatial/__init__.py
+++ b/src/neo4j/_spatial/__init__.py
@@ -60,10 +60,7 @@ class Point(t.Tuple[float, ...]):
         def z(self) -> float: ...
 
     def __new__(cls, iterable: t.Iterable[float]) -> Point:
-        # mypy issue https://github.com/python/mypy/issues/14890
-        return tuple.__new__(  # type: ignore[type-var]
-            cls, map(float, iterable)
-        )
+        return tuple.__new__(cls, map(float, iterable))
 
     def __repr__(self) -> str:
         return "POINT(%s)" % " ".join(map(str, self))

--- a/src/neo4j/time/__init__.py
+++ b/src/neo4j/time/__init__.py
@@ -416,8 +416,7 @@ class Duration(t.Tuple[int, int, int, int],  # type: ignore[misc]
         if not MIN_INT64 <= avg_total_seconds <= MAX_INT64:
             raise ValueError("Duration value out of range: %r",
                              tuple.__repr__((mo, d, s, ns)))
-        # mypy issue https://github.com/python/mypy/issues/14890
-        return tuple.__new__(cls, (mo, d, s, ns))  # type: ignore[type-var]
+        return tuple.__new__(cls, (mo, d, s, ns))
 
     def __bool__(self) -> bool:
         """Falsy if all primary instance attributes are."""

